test/Microsoft.Azure.ServiceBus.UnitTests/SenderReceiverClientTestBase.cs

```diff
@@ -157,5 +158,60 @@ public abstract class SenderReceiverClientTestBase
 
             var receivedMessages = await TestUtility.ReceiveMessagesAsync(messageReceiver, messageCount);
         }
+
+        protected async Task ScheduleMessagesAsyncTestCase(MessageSender messageSender, MessageReceiver messageReceiver, int messageCount)
+        {
+            var startTime = DateTime.UtcNow;
+            var scheduleTime = new DateTimeOffset(DateTime.UtcNow).AddSeconds(5);
+            TestUtility.Log($"Sending message with schedule time: {scheduleTime.UtcDateTime}");
+
+            var sequenceNumber =
+                await
+                    messageSender.ScheduleMessageAsync(
+                        new BrokeredMessage("Test") { MessageId = "randomId", Label = "randomLabel" }, scheduleTime);
+            TestUtility.Log($"Received sequence number: {sequenceNumber}");
+            Assert.True(sequenceNumber > 0);
+
+            TestUtility.Log("Sleeping for 5 seconds...");
+            Thread.Sleep(TimeSpan.FromSeconds(5));
+
+            var message = await messageReceiver.ReceiveAsync();
+
+            // Asserting using Math.Ceiling since TotalSeconds usually ends up being around 4.999 due to precision of
+            // the scheduleTime in requestMessage and responseMessage.
+            Assert.True(Math.Ceiling(message.ScheduledEnqueueTimeUtc.Subtract(startTime).TotalSeconds) >= 5);
+        }
+
+        protected async Task CancelScheduledMessagesAsyncTestCase(MessageSender messageSender, MessageReceiver messageReceiver, int messageCount)
+        {
+            var scheduleTime = new DateTimeOffset(DateTime.UtcNow).AddSeconds(20);
+            var brokeredMessage = new BrokeredMessage("Test1") { MessageId = Guid.NewGuid().ToString() };
+            TestUtility.Log(
+                $"Sending message with schedule time: {scheduleTime.UtcDateTime} and messageID {brokeredMessage.MessageId}");
+
+            var sequenceNumber = await messageSender.ScheduleMessageAsync(brokeredMessage, scheduleTime);
+            TestUtility.Log($"Received sequence number: {sequenceNumber}");
+            Assert.True(sequenceNumber > 0);
```

Comment:
Ok. Based on another reply there would be an exception thrown, which leads me to think this will change (removed check for sequence number)